lib/samples/show_viewshed_from_geoelement_in_scene/show_viewshed_from_geoelement_in_scene.dart

```diff
@@ -0,0 +1,288 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import 'dart:async';
+
+import 'package:arcgis_maps/arcgis_maps.dart';
+import 'package:arcgis_maps_sdk_flutter_samples/common/common.dart';
+import 'package:flutter/material.dart';
+import 'package:path_provider/path_provider.dart';
+
+class ShowViewshedFromGeoelementInScene extends StatefulWidget {
+  const ShowViewshedFromGeoelementInScene({super.key});
+
+  @override
+  State<ShowViewshedFromGeoelementInScene> createState() =>
+      _ShowViewshedFromGeoelementInSceneState();
+}
+
+class _ShowViewshedFromGeoelementInSceneState
+    extends State<ShowViewshedFromGeoelementInScene>
+    with SampleStateSupport {
+  // Create a controller for the scene view.
+  final _sceneViewController = ArcGISSceneView.createController();
+
+  // The graphic for the tank.
+  Graphic? _tankGraphic;
+
+  // Timer for animation.
+  Timer? _animationTimer;
+
+  // Waypoint for tank graphic.
+  ArcGISPoint? _waypoint;
+
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            ArcGISSceneView(
```

Comment:
It's customary to say "`// Add a scene view to the widget tree and set a controller.`" above this line